cpp/README.rst

```diff
@@ -57,7 +57,7 @@ Release build:
     $ mkdir build-release
     $ cd build-release
     $ cmake ..
-    $ make -j8       # if you have 8 CPU cores, otherwise adjust
+    $ make -j8       # if you have 8 CPU cores, otherwise adjust, use -j`nproc` for all cores
```

Comment:
Maybe we can unify the usage of ".. code-block:: shell"  ".. code-block::" and ".. code:: shell" in this file

Reply:
> Thanks! I unified it in both README of cpp and README of spark.
> 
> Has any other place where I can make it better?

Thanks for your contribution! This change looks good to me now.